File: backend/domain/notification.go
package domain

import "time"

type NotificationType string

const (
	NotificationTypeLike NotificationType = "like"
)

type Notification struct {
	ID        string           `json:"id"`
	UserID    string           `json:"user_id"`
	ActorID   string           `json:"actor_id"`
	Actor     *User            `json:"actor,omitempty"`
	Type      NotificationType `json:"type"`
	ArtworkID *string          `json:"artwork_id,omitempty"`
	ReadAt    *time.Time       `json:"read_at"`
	CreatedAt time.Time        `json:"created_at"`
}

type NotificationRepository interface {
	Create(n *Notification) error
	FindByUserID(userID string, limit, offset int) ([]*Notification, error)
	MarkAllAsRead(userID string) error
	CountUnread(userID string) (int, error)
}

type NotificationService interface {
	GetForUser(userID string, limit, offset int) ([]*Notification, error)
	MarkAllAsRead(userID string) error
	CountUnread(userID string) (int, error)
}

File: mobile/lib/models/notification.dart
import 'user.dart';

class AppNotification {
  final String id;
  final String userId;
  final String actorId;
  final User? actor;
  final String type;
  final String? artworkId;
  final DateTime? readAt;
  final DateTime createdAt;

  AppNotification({
    required this.id,
    required this.userId,
    required this.actorId,
    this.actor,
    required this.type,
    this.artworkId,
    this.readAt,
    required this.createdAt,
  });

  bool get isRead => readAt != null;

  factory AppNotification.fromJson(Map<String, dynamic> json) {
    return AppNotification(
      id: json['id'] ?? '',
      userId: json['user_id'] ?? '',
      actorId: json['actor_id'] ?? '',
      actor: json['actor'] != null ? User.fromJson(json['actor']) : null,
      type: json['type'] ?? '',
      artworkId: json['artwork_id'],
      readAt: json['read_at'] != null ? DateTime.parse(json['read_at']) : null,
      createdAt: json['created_at'] != null
          ? DateTime.parse(json['created_at'])
          : DateTime.now(),
    );
  }
}
